client/server/config/parser.js

```diff
@@ -66,7 +66,19 @@ module.exports = function ( configPath, defaultOpts ) {
 		data.features[ 'wpcom-user-bootstrap' ] = false;
 	}
 
-	const serverData = assign( {}, data, getDataFromFile( secretsPath ) );
+	const secrets = getDataFromFile( secretsPath );
+	if ( opts.env === 'desktop' && secretsPath === realSecretsPath ) {
+		const oauthClientId = secrets.desktop_oauth_client_id;
+		const oauthSecret = secrets.desktop_oauth_client_secret;
+		const oauthTokenEndpoint = secrets.desktop_oauth_token_endpoint;
+
+		data.oauth_client_id = oauthClientId !== undefined ? oauthClientId : false;
+		data.desktop_oauth_client_secret = oauthSecret !== undefined ? oauthSecret : false;
+		data.desktop_oauth_token_endpoint =
+			oauthTokenEndpoint !== undefined ? oauthTokenEndpoint : false;
+	}
+
+	const serverData = assign( {}, data, secrets );
```

Comment:
What's strange is logging `window.configData.desktop_oauth_client_secret` yields `undefined`. At least for the Desktop app, at this point the `secrets` file contains the `desktop_oauth_client_secret` and friends. 

I would expect that the assignment as-is would ensure that the desktop values from `secrets.json` would be available in the final object.

Reply:
Here we're looking at the code that runs on server. The data from `secrets.json` are available only on server and there, `config( 'desktop_oauth_client_secret' )` will return the config value.

`window.configData` is the config that's visible to the client in the browser. The server doesn't send the secrets to the browser, for anyone to see.

Look at the code in the [server config module](https://github.com/Automattic/wp-calypso/blob/master/client/server/config/parser.js#L69-L70) that sets the `serverData` and `clientData`. `serverData` contain the secrets, `clientData` don't.

The `serverData` are used to populate the server-side `config( flag )` and `config.isEnabled( flag )` implementations.

The `clientData` are sent in the HTML response as a script, visible to the browser as `window.clientData`. These data are then used to populate the client-side `config` object.